lib/fluent/plugin/in_tail.rb

```diff
@@ -494,9 +500,9 @@ def on_notify
             begin
               while true
                 if @buffer.empty?
-                  @io.read_nonblock(2048, @buffer)
+                  @io.readpartial(2048, @buffer)
```

Comment:
At least on Mac OS X and Linux., read_nonblock raises EOFError if IO is for a local file and the file reaches end-of-file. It doesn't raise Errno::EAGAIN or Errno::EWOULDBLOCK. Although it seems better to rescue Errno::EAGAIN and Errno::EWOULDBLOCK as well as EOFError.

If we aware of it, current `IOHandler#on_notify` is not a infinite loop because it returns immediately when file has no extra bytes.

`IOHandler#on_notify` is called by TimerWatcher (trigger using timer) as well as StatWatcher (trigger using Coolio::StatWatcher). When on_notify is called by TimerWatcher, file may have 0 bytes.

So, doesn't this change cause issue when 2 or more TimerWatcher is registered and one of them is blocked at IO#readpartial?